### PR TITLE
UO should ignore Kafka built-in special users - Closes #1371

### LINKED ISSUE #1371: Allowing addition of User:ANONYMOUS and/or more generic ACLs
I was looking to launch a hybrid authentication Kafka cluster with configuration something like:
```plain: {}
      external:
        authentication:
          type: tls
```

There are k8s filebeat pods which wish to write to Kafka `topicA` and these logs aren't sensitive per se. So we were looking to push without tls/auth to `plain` port.

While there is another workload which generate sensitive logs `topicB` are we are looking to push with `external` tls auth.

So I was trying to make:
- unauthenticated persons can push & fetch logs from `topicA` and not `topicB`
- while only authenticated (username: xyz) can `topicB`

The ideal configuration in non-operator world.
- `allow.everyone.if.no.acl.found: false`
- adding `User:xyz` with permission over `topicB`
- granting `User:ANONYMOUS` permission to `topicA`

Possible Solutions:
1. Making `User:ANONYMOUS` crd possible, if yes other authorization can work as expected for other users. Currently only lower case usernames are allowed. ([ANONYMOUS is Kafka's implementation](https://docs.confluent.io/current/kafka/authorization.html#user-names))
2. Making custom `kafka-acls.sh` commands possible after Zookeeper are up, this can make topic level ACLs addition possible as well.

Quick Chat [link on Slack](https://strimzi.slack.com/archives/CAA5B4TJ8/p1551085404072500) for context.

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

Currently, users cannot set manually ACL rules for users such as `User:ANONYMOUS` or `User:*` when UO is enabled because it would delete the ACL rules as a rules for non-existent user. At the same time, the UO cannot manage these rules because `ANONYMOUS` and `*` are not valid Kubernetes names.

This PR adds exception handling to ignore these special users (and just print an info log messages so that users are aware of it).

This should close #1371.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=HelmChartST#HelmChartST,MirrorMakerST#testMirrorMakerTlsAuthenticated,MirrorMakerST#testMirrorMakerTlsScramSha

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 0.13.0
+
+* Allow users to manually configure ACL rules for special Kafka users `*` and `ANONYMOUS` without them being deleted by the User Operator.
```
Comment (tombentley) [inline CHANGELOG.md]:
```suggestion
* Allow users to manually configure ACL rules (for example, using using `kafka-acls.sh`) for special Kafka users `*` and `ANONYMOUS` without them being deleted by the User Operator.
```

Comment (azapps) [reply CHANGELOG.md]:
```suggestion
* Allow users to manually configure ACL rules (for example, using `kafka-acls.sh`) for special Kafka users `*` and `ANONYMOUS` without them being deleted by the User Operator.
```
:stuck_out_tongue:

Comment (tombentley) [reply CHANGELOG.md]: Thanks!

**Pull request merged**